src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -658,29 +676,30 @@ protected override void AttachCore()
         {
             // Initialize auto sync timer.
             int interval = (int)TimeSpan.FromMinutes(AutosyncIntervalMinutes).TotalMilliseconds;
-            this.autosyncTimer = new Timer(async (o) =>
+            this.autoSyncTimer = new Timer((o) =>
             {
-                await this.ResyncAsync().ConfigureAwait(false);
+                if (this.cachedHeaders.Count == 0)
```

Comment:
This might be an issue if for some reason the node is stuck and the cache does not get cleared, I wonder if we should clear the cache if it ahs not changed for an X interval (lets discuss)